client/src/main/java/io/pravega/client/tables/impl/SegmentIteratorArgs.java

```diff
@@ -29,17 +30,58 @@
 @Builder
 class SegmentIteratorArgs {
     /**
-     * Optional. If specified, all items returned by {@link AsyncIterator#getNext()} will have {@link TableSegmentKey}s
-     * that begin with the specified prefix.
+     * The Table Segment Key to begin iteration at (inclusive).
      */
-    private final ByteBuf keyPrefixFilter;
+    @NonNull
+    private final ByteBuf fromKey;
+    /**
+     * The Table Segment Key to end the iteration at (inclusive). NOTE that not all Keys/Entries between
+     * {@link #getFromKey()} and {@link #getToKey()} may be returned - the result size may be capped at
+     * {@link #getMaxItemsAtOnce()}.
+     */
+    @NonNull
+    private final ByteBuf toKey;
     /**
      * The maximum number of items to return with each call to {@link AsyncIterator#getNext()}.
      */
     private final int maxItemsAtOnce;
+
     /**
-     * Optional. A continuation token that can be used to resume a previously interrupted iteration. This can be obtained
-     * by invoking {@link IteratorItem#getState()}. A null value will create an iterator that lists all keys.
+     * Creates a new {@link SegmentIteratorArgs} that is identical to this instance, but has a {@link #getFromKey()}
+     * which is the immediate successor of the given {@code lastKey}.
+     *
+     * @param lastKey The last returned key from an iteration of the {@link TableSegmentIterator}.
+     * @return The next {@link SegmentIteratorArgs} to use, or null of {@code lastKey} is null or if, as a result of this
+     * method's computations, the next {@link SegmentIteratorArgs} would have {@link #getFromKey()} exceed
+     * {@link #getToKey()} (which means a subsequent call to {@link TableSegmentIterator#getNext()} cannot possibly yield
+     * any result).
      */
-    private final IteratorState state;
+    SegmentIteratorArgs next(@Nullable ByteBuf lastKey) {
+        if (lastKey == null) {
+            // End of iteration.
+            return null;
+        }
+
+        final byte[] result = lastKey.copy().array();
+        int index = result.length - 1;
+        ByteBuf resultBuf = null;
+        while (index >= 0) {
+            int v = result[index] & 0xFF;
+            if (v >= 0xFF) {
```

Comment:
Because it can be > 0xFF this is the same as an equality check, in which case if it is equal to 0xFF then adding 1 would result in 0 anyway, so this part can be combined with the other branch. The if is only needed for the break, which can be done by checking if the value after increment is not 0.

Reply:
Thanks for the tip. I simplified it as per your suggestion.